### ISSUE TITLE
Question: How do i document an expected response without an 'object'

### ISSUE THREAD

**Issue by coderangerdan** [domaindrivendev/Swashbuckle#1076]
I return a response without an 'object' but by serialising some arbitrary values; how do i get Swashbuckle to output this response without generating a proper 'object'?

**Comment by heldersepu** [comment]
Do you have a Code sample?

**Comment by coderangerdan** [comment]
Not really, its serializing a sql query into JSON or XML, so its not determined by an object model as it cant be interpreted ahead of time; so I need to tell swashbuckle what a sample response will be

**Comment by heldersepu** [comment]
You can decorate your actions with ResponseType

```c#
[ResponseType(typeof(IEnumerable<string>))]
public IHttpActionResult Get()
{
    return Ok(new string[] { "value1", "value2" });
}
```
Is that what you are looking for?

**Comment by coderangerdan** [comment]
Not really as that just says the example response would be
`[
  "string"
]`

I want to document that the response could be (depending on Accepts type):
```
<channels>
    <channel>
        <id>2</id>
        <name>Test World</name>
    </channel>
    <channel>
        <id>28</id>
        <name>Test World 2</name>
    </channel>
</channels>
```
Or
```
{
  "channels": [
    {
      "id": 2,
      "name": "Test World",
    }.
    {
      "id": 28,
      "name": "Test World 2",
    }
    ]
}
```

**Comment by heldersepu** [comment]
Why not create a model that matches that and put in the ResponseType?

**Comment by coderangerdan** [comment]
I see, as an example, but not as an actual response; thats a good idea, thanks.

However my response is slightly different (for legacy reasons) to that which native WebApi would have generated; note my JSON response above adds an object property containing an array, whereas native webapi would serialize as just an array

**Comment by coderangerdan** [comment]
This looks like what I need so I will try it out. [Swashbuckle.Examples](https://mattfrear.com/2016/01/25/generating-swagger-example-requests-with-swashbuckle/)

**Comment by heldersepu** [comment]
I'm glad that you found something that works for you...
Since this issue is not a real issue with Swashbuckle, can you please close it.

**Comment by coderangerdan** [comment]
Hmm tried all day and i just cant get any joy out of either the default behaviour nor with the examples package.

### Attempt 1

I have changed my default JSON output to just be an array of objects to make things easier; and, I have defined a model:
```
    class Channel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Active { get; set; }
        public int DefaultTerritoryID { get; set; }
    }
```
And added a response type:
`[ResponseType( typeof( IEnumerable<Channel> ) )]`

I get the expected JSON response of:
```
[
  {
    "id": 0,
    "name": "string",
    "active": 0,
    "defaultTerritoryID": 0
  }
]
```

But the XML response is:
```
<Inline Model>
  <id>1</id>
  <name>string</name>
  <active>1</active>
  <defaultTerritoryID>1</defaultTerritoryID>
</Inline Model>
```
This appears to be a bug reported in [swagger-ui](https://github.com/swagger-api/swagger-ui/issues/2325)  and fixed but not yet merged into swashbuckle.

The same happens with:
`[SwaggerResponse( HttpStatusCode.OK, Type = typeof( IEnumerable<channel> ) )]`

### Attempt 2

So I then tried to define a wrapper class:
```
    class Channels
    {
        public List<Channel> Channel { get; set; }
    }
```
And changed my response to:
`[ResponseType( typeof( Channels ) )]`

Ok that gave a different result in that the XML response types have now disappeared and the JSON is now broken:
```
{
  "channel": [
    {
      "id": 0,
      "name": "string",
      "active": 0,
      "defaultTerritoryID": 0
    }
  ]
}
```

### Attempt 3

I tried one last thing and change Attempt 2 to use a different response:
`[SwaggerResponse( HttpStatusCode.OK, Type = typeof( Channels ) )]`

This adds back the XML response and is correct:
```
<channels>
  <channel>
    <id>1</id>
    <name>string</name>
    <active>1</active>
    <defaultTerritoryID>1</defaultTerritoryID>
  </channel>
</channels>
```

But the JSON is now incorrect:
```
{
  "channel": [
    {
      "id": 0,
      "name": "string",
      "active": 0,
      "defaultTerritoryID": 0
    }
  ]
}
```

So I just basically cant get any of it to be correct; what i want is a combination of both:
```
<channels>
  <channel>
    <id>1</id>
    <name>string</name>
    <active>1</active>
    <defaultTerritoryID>1</defaultTerritoryID>
  </channel>
</channels>

[
  {
    "id": 0,
    "name": "string",
    "active": 0,
    "defaultTerritoryID": 0
  }
]
```

Failing that, i just want to add an XML comment that describes my response exactly as i want it rather than the system trying to deduce it and get it all wrong all the time. I guess my only solution is to just add it into the <remarks> section unless someone can think of an alternative.
